### PR TITLE
Add tabs to job panel for active/completed/failed jobs

### PR DESCRIPTION
**Pull request by jakereps**
Adds tabs to cycle through Active, Completed, and Failed jobs.

- [x] Add tabs to job view "table"
- [x] Add reducer to handle tab changes/views
- [x] Add other job states/tracking to current reducer
- [x] Display proper view when tab changed
    - [x] Fill view with proper jobs from state
- [x] Job history interaction?

### CODE REVIEW

**Comment by ebolyen** [comment]
https://github.com/qiime2/qiime2/issues/55 has been "fixed" so you should be able to update this to use the `Job.uuid`

**Comment by jakereps** [comment]
Should be ready for review after testing. Switch it to use the `webContents` data passing and `setState()` (as there are no reducer/store actions being used on this solo page) to get the entire job object to the history window.

File: app/js/components/JobRow.jsx [hunk 1]
```diff
@@ -1,38 +1,27 @@
...
-                <span aria-hidden="true">&times;</span>
-            </button><br />
-            {data.message}
-        </div>
+            </a>
+        </td>
+        <td>
+            {data.started}
+        </td>
+        { data.timestamp ?
```
Comment (ebolyen) [inline app/js/components/JobRow.jsx]: This should be `data.finished`

File: app/js/reducers/index.js [hunk 2]
```diff
@@ -1,19 +1,21 @@
 import { combineReducers } from 'redux';
-import { routerReducer } from 'react-router-redux';
+import { routerReducer as routing } from 'react-router-redux';
```
Comment (ebolyen) [inline app/js/reducers/index.js]: :+1:

File: qiime_studio/api/v1.py [hunk 3]
```diff
@@ -149,35 +137,44 @@ def execute_workflow(plugin_name, workflow_name):
...
+    __JOBS[str(job.uuid)] = {
         'completed': False,
         'error': False,
-        'message': ''
+        'finished': None,
+        'started': now
     }
-    future_result.add_done_callback(toggle_completion)
+    future_result.add_done_callback(lambda future_result:
+                                    toggle_completion(future_result, job))
```
Comment (ebolyen) [inline qiime_studio/api/v1.py]: Nice use of `lambda`

File: app/js/components/JobHistory.jsx [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    constructor(props) {
+        super(props);
+        this.state = { job: {}, ...props };
+        this.order = ['UUID', 'Completed', 'Error', 'Started', 'Finished', 'stdout', 'stderr'];
+    }
+
+    componentWillMount() {
+        ipc.on('pass-job-data', (event, data) => {
+            this.setState({ job: data });
+        });
```
Comment (ebolyen) [inline app/js/components/JobHistory.jsx]: Someday we will be able to just have this in a shared store, but this is a good solution for now!

**Comment by ebolyen** [comment]
@jakereps One small bug, but ready for merge after that is fixed.

File: qiime_studio/api/v1.py (hunk 3 above)
Comment (jakereps) [inline qiime_studio/api/v1.py]: Thanks! Now that we were using the job info to store it in the dictionary it needed a way to get both pieces in the callback.

**Comment by jakereps** [comment]
Thanks, @ebolyen! Should be good after tests pass! Realized a way to clean up the massive stack of ifs and switches while doing the Artifacts/Visualizations tabs, so will probably refactor these tabs to match in that new PR.

**Comment by jakereps** [comment]
Ahh!!! It was failing from the merged #52, which still had the package.json with no lint pretest set up! I was looking everywhere for why it was saying something about a boolean in a spot that had no booleans. Fixed!

**Comment by ebolyen** [comment]
Looks great, thanks @jakereps!

**Pull request merged**